obs-studio-server/source/nodeobs_service.cpp

```diff
@@ -2097,50 +2104,76 @@ void OBS_service::UpdateRecordingSettings()
 	}
 }
 
+void OBS_service::releaseEncoders()
+{
+	obs_encoder_release(audioStreamingEncoder);
+	obs_encoder_release(audioRecordingEncoder);
+	obs_encoder_release(videoStreamingEncoder);
+	obs_encoder_release(videoRecordingEncoder);
+}
+
 obs_encoder_t* OBS_service::getStreamingEncoder(void)
 {
+	obs_encoder_addref(videoStreamingEncoder);
 	return videoStreamingEncoder;
 }
 
 void OBS_service::setStreamingEncoder(obs_encoder_t* encoder)
 {
-	if (!videoStreamingEncoder)
-		obs_encoder_release(videoStreamingEncoder);
+	obs_encoder_release(videoStreamingEncoder);
```

Comment:
In case when `videoStreamingEncoder` is the same as `encoder`. `obs_encoder_release` can destroy object and `obs_encoder_addref` be applied on destroyed object. 
Currently it can be non issue as no one yet call `setStreamingEncoder`  to set same encoder object. Or object still have few references and not destroyed right after `obs_encoder_release`. But when other code change this can become an issue. 
Or if someone call `setStreamingEncoder( getStreamingEncoder() )` it just increase ref count instead of doing nothing.

Reply:
I will add some checks to prevent it from happening, not sure if an assert would be better to proper indicate invalid cod structure.

Reply:
A case like `setStreamingEncoder(getStreamingEncoder() )` should never happen since the user is responsible to release any pointer that he got by calling `getxxxxxxEncoder()`, a fix for this would be using RAII.